From 427ff5a1bb80de443539cce070ae2c6721f864ce Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Mon, 24 Oct 2011 21:26:59 +0800
Subject: [PATCH 0473/2463] ENGR00160667 mx53 ard: fix ethernet unwork

1) use smsc911x instead of smc911x
The smsc911x is supposed to replace the old smc911x driver.
2) fix the wrong chip select address

Signed-off-by: Dong Aisheng <b29396@freescale.com>
---
 arch/arm/configs/imx5_defconfig       |    7 ++++---
 arch/arm/plat-mxc/include/mach/mx53.h |    4 ++--
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/arch/arm/configs/imx5_defconfig b/arch/arm/configs/imx5_defconfig
index be5f240..9c2687e 100755
--- a/arch/arm/configs/imx5_defconfig
+++ b/arch/arm/configs/imx5_defconfig
@@ -1,7 +1,7 @@
 #
 # Automatically generated make config: don't edit
 # Linux/arm 2.6.38 Kernel Configuration
-# Sat Oct  8 15:59:45 2011
+# Mon Oct 24 21:25:30 2011
 #
 CONFIG_ARM=y
 CONFIG_HAVE_PWM=y
@@ -788,12 +788,13 @@ CONFIG_MDIO_BITBANG=y
 CONFIG_MDIO_GPIO=y
 CONFIG_NET_ETHERNET=y
 # CONFIG_AX88796 is not set
-CONFIG_SMC91X=y
+# CONFIG_SMC91X is not set
 # CONFIG_DM9000 is not set
 # CONFIG_ENC28J60 is not set
 # CONFIG_ETHOC is not set
 # CONFIG_SMC911X is not set
-# CONFIG_SMSC911X is not set
+CONFIG_SMSC911X=y
+# CONFIG_SMSC911X_ARCH_HOOKS is not set
 # CONFIG_DNET is not set
 # CONFIG_IBM_NEW_EMAC_ZMII is not set
 # CONFIG_IBM_NEW_EMAC_RGMII is not set
diff --git a/arch/arm/plat-mxc/include/mach/mx53.h b/arch/arm/plat-mxc/include/mach/mx53.h
index 2559437..1bcddd6 100755
--- a/arch/arm/plat-mxc/include/mach/mx53.h
+++ b/arch/arm/plat-mxc/include/mach/mx53.h
@@ -164,8 +164,8 @@
  */
 #define MX53_CSD0_BASE_ADDR		0x90000000
 #define MX53_CSD1_BASE_ADDR		0xA0000000
-#define MX53_CS0_BASE_ADDR		0xB0000000
-#define MX53_CS1_BASE_ADDR		0xB8000000
+#define MX53_CS0_BASE_ADDR		0xF0000000
+#define MX53_CS1_BASE_ADDR		0xF4000000
 #define MX53_CS2_BASE_ADDR		0xC0000000
 #define MX53_CS3_BASE_ADDR		0xC8000000
 #define MX53_CS4_BASE_ADDR		0xCC000000
-- 
1.7.7.4

